scripts/make_release.py

```diff
@@ -15,43 +14,31 @@
 END_HEADER_MAGIC = b"@@END_HEADER@@"
 
 
-def get_artifacts_zip(artifacts, artifact_name):
-    r = requests.get(
-        artifacts[artifact_name]["archive_download_url"],
-        headers={"Authorization": f"token {token}"},
-        stream=True
-    )
-    r.raise_for_status()
-    decoding_classes = {
-        "application/zip": zipfile.ZipFile
-    }
-    return decoding_classes[r.headers['Content-Type']](BytesIO(r.content))
-
-
-def get_version(version, installer_metadata):
-    if version is None:
-        # If no version is given, convert the build version number to a release version
-        version = Version(installer_metadata["VER"])
-        version = ".".join(map(str, version.release))
-    version = Version(version)
-
-    if version.is_prerelease:
-        # Bump the pre-release digit
-        next_version = [
-            ".".join(map(str, version.release)),
-            "".join(map(str, version.pre[:-1] + (version.pre[-1] + 1,))),
-        ]
-    else:
-        # Bump the least significant digit
-        next_version = [
-            ".".join(map(str, version.release[:-1] + (version.release[-1] + 1,))),
-            "a1"
-        ]
-
-    return str(version), "".join(next_version)
-
-
 def main(run_id=None, requested_version=None, workflow_fn="build-and-test.yml"):
+    """Make a new release of DIRACOS2 based on a GitHub Actions CI run
+
+    Releases are made by:
+    1. Download the artefact from a GitHub Actions CI run
+    2. Editing it in place to change the version number
+    3. Creating a pre-release in GitHub
+    4. Uploading release artefacts to the new pre-release
+    5. If the version number is not a pre-release: converting the pre-release to a full release
+       causing the "latest" release to update
+    6. Editing the current "main" branch to be one version higher than the current latest release
+
+    Authentication is performed using the GitHub API token which is automatically made available
+    to all GitHub Actions jobs.
+
+    Parameters
```

Comment:
Can we just use the normal sphinx documentation syntax for parameters and returns?

Reply:
Definitely, can you point me to an example as I can never remember which style is which

Reply:
Thanks!
Like here (including newline before), `:param theType name:` or `:param name:`, `:type name: theType or thisType`
https://github.com/DIRACGrid/DIRAC/blob/841e68f1880fa72c74d27a4876ba2ee2c4f72a72/TransformationSystem/Utilities/ReplicationTransformation.py#L23-L37